### DISCUSSION TITLE
Button without state - how?

### DISCUSSION THREAD

**Discussion by kpreobra** [RavenSystem/esp-homekit-devices#1689] [MEPLHAA Scripts]
Hello.
How can i make a switch as an inching button without state remembering?
I made this:
{"t":1,"0":{"r":[{"g":5,"v":0}]},"1":{"r":[{"g":5,"v":1,"i":0.5}]},"b":[{"g":9,"t":0}]}
it inching after 0,5 sec when i switch it on, but state remains "on", how can i get the state off too? Can i use just button for this case? End point - i need a button in home app, that switches relay for 0,5 second only, without state.

**Comment by mariusvasss** [comment]
{"t":1,"0":{"r":[{"g":5,"v":0}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9,"t":0}],"i":0.5}]}

**Reply by kpreobra** [reply]
Thanks a lot, it works!

**Reply by mariusvasss** [reply]
You welcome.

**Reply by kpreobra** [reply]
Hello, and is there any possibility to get state of device, attached to sonoff? for example if i open garage gates manually or from remote contol, can it be possible?

**Reply by xrust83** [reply]
exists.  called a limit switch

**Reply by kpreobra** [reply]
so, if i just connect a relay, which operating like "on" when gates are opened and "off" when gates are closed - it will be working without any additional settings?

**Reply by xrust83** [reply]
Hard to say.  the world is full of surprises and I alone think that you are a fairly educated technical person and say yes, but it may turn out that you have come up with an answer for yourself no or even worse categorically no.  it all depends on different factors.  we have a smart gate system and you connect a relay to a block of buttons that simulate pressing them, or you have a sonoff dual that is simply connected to the motor and can change its direction of movement.  Perhaps you have something similar to the heating battery valve actuator, which opens when 220V is supplied and closes when it is turned off.  provide a diagram, or a model of the finished device, and if someone has done this, they will share advice with you, and the question you asked cannot have an answer.

**Reply by kpreobra** [reply]
i have a sonoff dual connected to gates control, it simulates button pressing (inching for 0,5 sec), like i pressed button on the remote control.

**Reply by xrust83** [reply]
in this case, you are not rationally using sonoff dual for this work, it would be worth wemos d1 mini for $ 1.5.  in addition, your device is forwarded to HomeKit as two simple switches.  it will work accordingly, but there can be no talk of any display of the state of the gate.  For HomeKit, these are no more than two switches.

**Reply by kpreobra** [reply]
thanks, but i need garage door device in homekit to use siri suggestions on the carplay. cal wemos seem as garage door device?

**Reply by xrust83** [reply]
When writing json you specified for your device type = 1, there is a normal switch. For garage doors you need type = 40
https://github.com/RavenSystem/esp-homekit-devices/wiki/Service-Types
